src/librustdoc/clean/mod.rs

```diff
@@ -1824,10 +1824,19 @@ impl Clean<Visibility> for ty::Visibility {
     }
 }
 
+crate fn struct_type_from_def(vdata: &hir::VariantData<'_>) -> StructType {
+    use StructType::*;
+    match *vdata {
+        hir::VariantData::Struct(..) => Plain,
+        hir::VariantData::Tuple(..) => Tuple,
+        hir::VariantData::Unit(..) => Unit,
+    }
+}
```

Comment:
Could we actually go even further and use [`CtorKind`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/def/enum.CtorKind.html) instead of adding our own type? They're bijective I'm pretty sure, you can use [`from_hir`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/def/enum.CtorKind.html#method.from_hir) in place of this function.

Reply:
It seems that `inline.rs` is also converting from `CtorKind` to `StructKind`, so yeah I can switch to just using that